Engine/src/Engine/UserInterface.cpp

```diff
@@ -0,0 +1,101 @@
+#include "UserInterface.h"
+
+ImGuiIO* UserInterface::s_IO = nullptr;
+ImGuiStyle* UserInterface::s_style = nullptr;
+
+std::map<std::string, ImFont*> UserInterface::s_fonts;
+
+//define the ImGUI vec4s for each colour
+ImVec4 UserInterface::red = ImVec4(1.0f, 0.0f, 0.0f, 1.0f);
+ImVec4 UserInterface::green = ImVec4(0.0f, 1.0f, 0.0f, 1.0f);
+ImVec4 UserInterface::blue = ImVec4(0.0f, 0.0f, 1.0f, 1.0f);
+ImVec4 UserInterface::white = ImVec4(1.0f, 1.0f, 1.0f, 1.0f);
+
+//initializes a GLFWwindow that can be used with ImGUI
+bool UserInterface::initialize(GLFWwindow* window)
+{
+	ImGui::CreateContext();
+	ImGui_ImplOpenGL3_Init("#version 330");
+	ImGui_ImplGlfw_InitForOpenGL(window, true);
+
+	//TODO - Reference for style
+	//https://www.unknowncheats.me/forum/c-and-c-/189635-imgui-style-settings.html
+	s_style = &ImGui::GetStyle();
+
+	s_IO = &ImGui::GetIO();
+
+	return true;
+}
+
+void UserInterface::shutdown()
+{
+
+}
+
+//creates new ImGUI frame (window)
+void UserInterface::startUI()
+{
+	ImGui_ImplOpenGL3_NewFrame();
+	ImGui_ImplGlfw_NewFrame();
+	ImGui::NewFrame();
+}
+
+void UserInterface::endUI()
+{
+	ImGui::Render();
+	ImGui_ImplOpenGL3_RenderDrawData(ImGui::GetDrawData());
+}
+
+//loads a font, returns true if the font is loaded correctly
```

Comment:
Future consideration: Do these provide UI elements we can use in the game, and not just the engine? If so that may save us some work.

Reply:
Yes, the UI tools are designed to be reusable throughout game and engine.